x-pack/plugins/reporting/server/export_types/common/pdf/worker.js

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+require('../../../../../../../src/setup_node_env');
```

Comment:
This is just for code transpilation, correct? It implies the worker process isn't a full instance of Kibana, right?

Reply:
Yeah, exactly, this hooks into the node require environment and intercepts, for example, TS files, compiles and returns the compiled version. This does mean we are doing that legwork in the worker though...